lib/SPIRV/SPIRVToOCL12.cpp

```diff
@@ -85,12 +85,25 @@ void SPIRVToOCL12::visitCallSPIRVMemoryBarrier(CallInst *CI) {
   mutateCallInstOCL(
       M, CI,
       [=](CallInst *, std::vector<Value *> &Args) {
-        auto GetArg = [=](unsigned I) {
-          return cast<ConstantInt>(Args[I])->getZExtValue();
-        };
-        auto Sema = mapSPIRVMemSemanticToOCL(GetArg(1));
-        Args.resize(1);
-        Args[0] = getInt32(M, Sema.first);
+        if (isa<ConstantInt>(Args[1])) {
+          auto GetArg = [=](unsigned I) {
+            return cast<ConstantInt>(Args[I])->getZExtValue();
+          };
+          auto Sema = mapSPIRVMemSemanticToOCL(GetArg(1));
+          Args.resize(1);
+          Args[0] = getInt32(M, Sema.first);
+        } else {
+          CallInst *TransCall = dyn_cast<CallInst>(Args[1]);
+          Function *F = TransCall ? TransCall->getCalledFunction() : nullptr;
+          if (F && F->getName().equals(kSPIRVName::TranslateOCLMemScope)) {
+            Args[0] = TransCall->getArgOperand(0);
+          } else {
+            Args[0] = getOrCreateSwitchFunc(kSPIRVName::TranslateSPIRVMemScope,
```

Comment:
This is not the same as `mapSPIRVMemSemanticToOCL`.

Reply:
Fixed, thanks!